nntrainer/layers/layer_devel.h

```diff
@@ -31,14 +31,14 @@ namespace ml::train {
 class Layer;
 }
 
+enum class ExportMethods;
```

Comment:
Why do you need to declare ```ExportMethods``` in different headers?

Can you define it in a single header file (e.g., ```nntrainer-api.common.h```) and include this file directly or indirectly from the required headers/sources?

Reply:
Besides, as long as ExportMethods is no longer using "namespace", you don't need enum class. For such enum, ```typedef enum ...``` is enough.